app/dashboard/_components/ActivityFeed/ActivityFeed.tsx

```diff
@@ -1,18 +1,28 @@
+'use client';
+
+import type { Events } from '@prisma/client';
+import { useQuery } from '@tanstack/react-query';
+import { useSearchParams } from 'next/navigation';
 import { DataTableSkeleton } from '~/components/data-table/data-table-skeleton';
 import ActivityFeedTable from './ActivityFeedTable';
-import { Suspense } from 'react';
-import { searchParamsCache } from './searchParamsCache';
-import { getActivities } from '~/queries/activityFeed';
 
-export default function ActivityFeed() {
-  const searchParams = searchParamsCache.all();
-  const activitiesPromise = getActivities(searchParams);
+export const ActivityFeed = () => {
+  const params = useSearchParams();
+
+  const { isPending, data } = useQuery({
+    queryKey: ['activityFeed', params.toString()],
```

Comment:
Should the params be hashed for the queryKey?

Reply:
Not in this case. Unlike with the next cache, there aren't any concerns about key length, because tanstack query hashes internally itself: https://tanstack.com/query/latest/docs/framework/react/guides/query-keys#query-keys-are-hashed-deterministically

The only requirement is that the key be serialisable, which SearchParams isn't by default, but is when converted to a string.